test/slam_manifold_preintegration_update_test.cc

```diff
@@ -0,0 +1,383 @@
+  // symbolic formulation of ImuManifoldPreintegrationUpdate. These derivatives can be
+  // be automatically computed using tangent_jacobians, but the expressions can also be obtained
+  // by hand. For whatever reason (likely having to do with CSE) the handwritten expressions use
+  // fewer ops. To verify that the handwritten implementations are correct, we compare them to
+  // the version of ImuManifoldPreintegrationUpdate using the automatic derivatives.
+  using M99 = Eigen::Matrix<double, 9, 9>;
+  std::mt19937 gen(1104);
+
+  for (int i_ = 0; i_ < 10; i_++) {
+    // Generate sample inputs to preintegration update functions
+    const sym::Rot3d DR = sym::Rot3d::Random(gen);
+    const Eigen::Vector3d Dv = sym::Random<Eigen::Vector3d>(gen);
+    const Eigen::Vector3d Dp = sym::Random<Eigen::Vector3d>(gen);
+
+    const M99 covariance =
+        (sym::Random<M99>(gen) + 10 * M99::Identity()).selfadjointView<Eigen::Lower>();
+    const Eigen::Matrix3d DR_D_gyro_bias = sym::Random<Eigen::Matrix3d>(gen);
+    const Eigen::Matrix3d Dv_D_accel_bias = sym::Random<Eigen::Matrix3d>(gen);
+    const Eigen::Matrix3d Dv_D_gyro_bias = sym::Random<Eigen::Matrix3d>(gen);
+    const Eigen::Matrix3d Dp_D_accel_bias = sym::Random<Eigen::Matrix3d>(gen);
+    const Eigen::Matrix3d Dp_D_gyro_bias = sym::Random<Eigen::Matrix3d>(gen);
+
+    const Eigen::Vector3d accel = sym::Random<Eigen::Vector3d>(gen);
+    const Eigen::Vector3d gyro = sym::Random<Eigen::Vector3d>(gen);
+
+    const double dt = 1.24;
+
+    const Eigen::Vector3d accel_bias = sym::Random<Eigen::Vector3d>(gen);
+    const Eigen::Vector3d gyro_bias = sym::Random<Eigen::Vector3d>(gen);
+    const Eigen::Vector3d accel_cov = sym::Random<Eigen::Vector3d>(gen);
+    const Eigen::Vector3d gyro_cov = sym::Random<Eigen::Vector3d>(gen);
+
+    // Calculate outputs of handwritten derivative version
+    Eigen::Matrix<double, 9, 9> new_covariance;
+    Eigen::Matrix3d new_DR_D_gyro_bias;
+    Eigen::Matrix3d new_Dv_D_accel_bias;
+    Eigen::Matrix3d new_Dv_D_gyro_bias;
+    Eigen::Matrix3d new_Dp_D_accel_bias;
+    Eigen::Matrix3d new_Dp_D_gyro_bias;
+
+    sym::ImuManifoldPreintegrationUpdate<double>(
+        DR, Dv, Dp, covariance, DR_D_gyro_bias, Dv_D_accel_bias, Dv_D_gyro_bias, Dp_D_accel_bias,
+        Dp_D_gyro_bias, accel_bias, gyro_bias, accel_cov, gyro_cov, accel, gyro, dt,
+        sym::kDefaultEpsilond, nullptr, nullptr, nullptr, &new_covariance, &new_DR_D_gyro_bias,
+        &new_Dv_D_accel_bias, &new_Dv_D_gyro_bias, &new_Dp_D_accel_bias, &new_Dp_D_gyro_bias);
+
+    // Calculate outputs of auto derivative version
+    Eigen::Matrix<double, 9, 9> new_covariance_auto;
+    Eigen::Matrix3d new_DR_D_gyro_bias_auto;
+    Eigen::Matrix3d new_Dv_D_accel_bias_auto;
+    Eigen::Matrix3d new_Dv_D_gyro_bias_auto;
+    Eigen::Matrix3d new_Dp_D_accel_bias_auto;
+    Eigen::Matrix3d new_Dp_D_gyro_bias_auto;
+
+    sym::ImuManifoldPreintegrationUpdate<double>(
```

Comment:
This function is the non-auto one :)

Explains why they're so close, I was a bit surprised

Reply:
Thanks for catching this. We discussed this in person, but it turns out the test would have been failing (severely) had I called the other function correctly. Specifically, it turned out that the hand-written symbolic implementation of `right_jacobian` added a term it should have been subtracting, causing the problems. With that fixed, the difference between the two (using the frobenius norm) was on the order of `10-8`.